gcovr/gcov.py

```diff
@@ -183,14 +183,14 @@ def guess_source_file_name(
 
 def guess_source_file_name_via_aliases(gcovname, currdir, data_fname):
     common_dir = commonpath([data_fname, currdir])
-    fname = realpath(os.path.join(common_dir, gcovname))
+    fname = os.path.abspath(os.path.join(common_dir, gcovname))
```

Comment:
(Remark) This change looks like a significant deviation from the previous semantics, which might cause some previous workflows to break. But from my perspective it's 50/50 whether this breaks stuff or fixes long-standing problems, so I'm OK with this change.